src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -909,6 +909,15 @@ internal int CalculateSyntaxOffsetInSynthesizedConstructor(int position, SyntaxT
                 return syntaxOffset;
             }
 
+            if (localKind == SynthesizedLocalKind.InstrumentationPayload)
```

Comment:
Preliminary testing indicates that that will work. I'm much happier without the added parameters, and the test is actually more accurate.